@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -710,15 +710,17 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
         featureId: number | undefined,
         env: MapEnv
     ): void {
-        const meshBuffers = this.findOrCreateMeshBuffers(techniqueIndex, GeometryType.Polygon);
+        const isExtruded = isExtrudedPolygonTechnique(technique);
+
+        const geometryType = isExtruded ? GeometryType.ExtrudedPolygon : GeometryType.Polygon;
+        const meshBuffers = this.findOrCreateMeshBuffers(techniqueIndex, geometryType);
```

Comment:
Also, why does the method applyPolygonTechnique extract the techniqueIndex from the technique and pass it in separately.